streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamAggregate.java

```diff
@@ -77,43 +83,51 @@ public void init(final org.apache.kafka.streams.processor.ProcessorContext conte
         }
 
         @Override
-        public void process(final K key, final V value) {
+        public void process(final Record<KIn, VIn> record) {
             // If the key or value is null we don't need to proceed
-            if (key == null || value == null) {
-                LOG.warn(
-                    "Skipping record due to null key or value. key=[{}] value=[{}] topic=[{}] partition=[{}] offset=[{}]",
-                    key, value, context().topic(), context().partition(), context().offset()
-                );
+            if (record.key() == null || record.value() == null) {
+                if (context.recordMetadata().isPresent()) {
```

Comment:
Huh, it's weird that the ContextualProcessor exposes both the `context` field and the `context()` getter for the same field. Would you mind making that field private so that all references go through the method?

```suggestion
                if (context().recordMetadata().isPresent()) {
```